commands/plugin-install.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+const spawn = require('child-process-ext/spawn');
+const fsp = require('fs').promises;
+const fse = require('fs-extra');
+const path = require('path');
+const _ = require('lodash');
+const cjsResolve = require('ncjsm/resolve/sync');
+const isPlainObject = require('type/plain-object/is');
+const yaml = require('js-yaml');
+const cloudformationSchema = require('@serverless/utils/cloudformation-schema');
+const log = require('@serverless/utils/log');
+const ServerlessError = require('../lib/serverless-error');
+const yamlAstParser = require('../lib/utils/yamlAstParser');
+const npmCommandDeferred = require('../lib/utils/npm-command-deferred');
+const {
+  getPluginInfo,
+  getServerlessFilePath,
+  validate,
+} = require('../lib/commands/plugin-management');
+
+module.exports = async ({ configuration, serviceDir, configurationFilename, options }) => {
+  validate({ serviceDir });
+
+  const pluginInfo = getPluginInfo(options.name);
+  const pluginName = pluginInfo.name;
+  const pluginVersion = pluginInfo.version || 'latest';
+  const configurationFilePath = getServerlessFilePath({ serviceDir, configurationFilename });
+
+  const context = { configuration, serviceDir, configurationFilePath, pluginName, pluginVersion };
+  if (await isInstalled(context)) {
```

Comment:
Sorry for the second thought, but I just realized that probably it'll be best to follow similar logic as `npm install` does.

So in all cases follow with installation.

Thing is that user may e.g. upgrade the plugin, and use `sls plugin install` command simply to upgrade. It should remain working (especially in case of standalone where user may not have `npm` globally installed.

So in light of that I would simply remove this condition, then follow with `npm install` in all cases, and add to the `serverless.yml` only if we see it's not there already

Reply:
No problem :) I fully agree with this new idea. Let me apply this!

> add to the serverless.yml only if we see it's not there already

FYI, it would be already done by this line:
https://github.com/serverless/serverless/blob/f32b6c346ef34bff7f64f7154dcd3c5ca8ccf729/commands/plugin-install.js#L90